plugins/web-worker-offloading/hooks.php

```diff
@@ -0,0 +1,138 @@
+	if ( false === $partytown_js ) {
+		return;
+	}
+
+	wp_register_script(
+		'web-worker-offloading',
+		'',
+		array(),
+		WEB_WORKER_OFFLOADING_VERSION,
+		array( 'in_footer' => false )
+	);
+
+	wp_add_inline_script(
+		'web-worker-offloading',
+		sprintf(
+			'window.partytown = %s;',
+			wp_json_encode( wwo_get_configuration() )
+		),
+		'before'
+	);
+
+	wp_add_inline_script( 'web-worker-offloading', $partytown_js );
+}
+add_action( 'wp_enqueue_scripts', 'wwo_init' );
+
+/**
+ * Marks scripts with `web-worker-offloading` dependency as async.
+ *
+ * This is needed because scripts offloaded to a worker thread can be considered async. However, they may include `before` and `after` inline
+ * scripts that need sequential execution. Once marked as async, `filter_eligible_strategies()` determines if the
+ * script is eligible for async execution. If so, it will be offloaded to the worker thread.
+ *
+ * @since n.e.x.t
+ *
+ * @param string[]|mixed $script_handles Array of script handles.
+ *
+ * @return string[] Array of script handles.
+ */
+function wwo_update_script_strategy( $script_handles ): array {
+	$script_handles = array_intersect( (array) $script_handles, array_keys( wp_scripts()->registered ) );
+	foreach ( $script_handles as $handle ) {
+		if ( in_array( 'web-worker-offloading', wp_scripts()->registered[ $handle ]->deps, true ) ) {
+			wp_script_add_data( $handle, 'strategy', 'async' );
+		}
+	}
+
+	return $script_handles;
+}
+add_filter( 'print_scripts_array', 'wwo_update_script_strategy' );
+
+/**
+ * Updates script type for handles having `web-worker-offloading` as dependency.
+ *
+ * @since n.e.x.t
+ *
+ * @param string|mixed $tag    Script tag.
+ * @param string       $handle Script handle.
+ *
+ * @return string Script tag with type="text/partytown" for eligible scripts.
+ */
+function wwo_update_script_type( $tag, string $handle ): string {
+	if (
+		is_string( $tag ) &&
+		array_key_exists( $handle, wp_scripts()->registered ) &&
+		in_array( 'web-worker-offloading', wp_scripts()->registered[ $handle ]->deps, true )
+	) {
+		$html_processor = new WP_HTML_Tag_Processor( $tag );
+
+		while ( $html_processor->next_tag( array( 'tag_name' => 'SCRIPT' ) ) ) {
+			if ( $html_processor->get_attribute( 'id' ) === "{$handle}-js" ) {
+				if ( null === $html_processor->get_attribute( 'async' ) ) {
+					_doing_it_wrong(
+						'wwo_update_script_type',
+						esc_html(
+							sprintf(
+								/* translators: %s: script handle */
+								__( 'Unable to offload "%s" script to a worker. Script will continue to load in the main thread.', 'web-worker-offloading' ),
+								$handle
+							)
+						),
+						'Web Worker Offloading n.e.x.t'
+					);
+				} else {
+					$html_processor->set_attribute( 'type', 'text/partytown' );
+					$html_processor->remove_attribute( 'async' );
+					$tag = $html_processor->get_updated_html();
+				}
+			}
```

Comment:
Minor suggestion to reduce nesting:

```suggestion
			if ( $html_processor->get_attribute( 'id' ) !== "{$handle}-js" ) {
				continue;
			}
			if ( null === $html_processor->get_attribute( 'async' ) ) {
				_doing_it_wrong(
					'wwo_update_script_type',
					esc_html(
						sprintf(
							/* translators: %s: script handle */
							__( 'Unable to offload "%s" script to a worker. Script will continue to load in the main thread.', 'web-worker-offloading' ),
							$handle
						)
					),
					'Web Worker Offloading n.e.x.t'
				);
			} else {
				$html_processor->set_attribute( 'type', 'text/partytown' );
				$html_processor->remove_attribute( 'async' );
				$tag = $html_processor->get_updated_html();
			}
```